### PR TITLE
Offset limit pagination 

### PR DESCRIPTION
**Pull request by kayladnls**

### CODE REVIEW

File: spec/Hooks/Pagination/OffsetLimitPaginationSpec.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    public function it_assigns_offset_to_request_as_integer()
+    {
+        $request = $this->process(Request::create('123/yolo?limit=20&offset=40', 'GET'));
+
+        $request->offset->shouldBe(40);
+    }
+
+    public function it_assigns_default_limit_of_zero_when_none_given()
```
Comment (kayladnls) [inline spec/Hooks/Pagination/OffsetLimitPaginationSpec.php]: This function is incorrectly named.

Comment (traviskaufman) [inline spec/Hooks/Pagination/OffsetLimitPaginationSpec.php]: whoops! Fixed.

File: src/Hooks/Pagination/OffsetLimitPagination.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+     * @return int
+     */
+    private function coerce($param, $param_name)
+    {
+        if (!$param) {
+            return;
+        }
+
+        if (is_numeric($param)) {
+            $coerced = intval($param);
```
Comment (kayladnls) [inline src/Hooks/Pagination/OffsetLimitPagination.php]: Why are we casting to an int and then to a float?

File: src/Hooks/Pagination/SinglePaginationHook.php [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?php namespace Refinery29\Piston\Hooks\Pagination;
+
+use League\Route\Http\Exception\BadRequestException;
+
+trait SinglePaginationHook
+{
+    public function ensureNotPreviousPaginated($request)
```
Comment (kayladnls) [inline src/Hooks/Pagination/SinglePaginationHook.php]: I know I did this, but can we change it to `ensureNotPreviouslyPaginated($request)`

File: src/Hooks/Pagination/OffsetLimitPagination.php (hunk 2 above)
Comment (traviskaufman) [inline src/Hooks/Pagination/OffsetLimitPagination.php]: in order to check whether or not the value passed was a float, per our conversation that passing a float should throw so `floatval(5) == intval(5)` will be truthy, yet `floatval(5.1) == intval(5.1)` will not.

Comment (traviskaufman) [inline src/Hooks/Pagination/OffsetLimitPagination.php]: is there a better way to accomplish this?

File: src/Hooks/Pagination/SinglePaginationHook.php (hunk 3 above)
Comment (traviskaufman) [inline src/Hooks/Pagination/SinglePaginationHook.php]: For sure.

File: src/Hooks/Pagination/OffsetLimitPagination.php (hunk 2 above)
Comment (kayladnls) [inline src/Hooks/Pagination/OffsetLimitPagination.php]: You could skip the `is_numeric()` in favor of an `is_int()` call, and that should fix it. Thoughts?

**Pull request merged**